maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -2369,7 +2369,21 @@ private Artifact getSurefireBooterArtifact()
      */
     File getSurefireTempDir()
     {
-        return IS_OS_WINDOWS ? createSurefireBootDirectoryInTemp() : createSurefireBootDirectoryInBuild();
+        File result = IS_OS_WINDOWS ? createSurefireBootDirectoryInTemp() : createSurefireBootDirectoryInBuild();
+        try
+        {
+            File canonical = result.getCanonicalFile();
+            if ( !result.equals( canonical ) )
+            {
+                logger.debug( "Canonicalized tempDir path " + result + " to " + canonical );
```

Comment:
Let's use simple quotation marks as it is down with IOException which avoids white spaces:
`logger.debug( "Canonicalized surefireTempDir path '" + result + "' to '" + canonical + "'." )`

Reply:
Good idea @Tibor17